src/WebJobs.Script/Config/FunctionsHostingConfigOptionsSetup.cs

```diff
@@ -33,6 +29,23 @@ public void Configure(FunctionsHostingConfigOptions options)
                     }
                 }
             }
+
+            // Restrict non-critial logs post-configuration/startup
+            ConfigureHostLogs(options);
+        }
+
+        private void ConfigureHostLogs(FunctionsHostingConfigOptions options)
```

Comment:
We should move the logic to honor the feature flag and set the filtered prefixes elsewhere to avoid expanding the responsibilities of the `FunctionsHostingConfigOptionsSetup`